src/buffer.rs

```diff
@@ -40,6 +40,7 @@ pub(crate) struct Buffer {
     decorations: Vec<Decoration>,
     selection_set_history: History<SelectionSet>,
     dirty: bool,
+    user: bool,
```

Comment:
Let's make this clearer by:
1. Renaming `user` to `opened_by`
2. Use `enum BufferOpenedBy {Lsp, User}`

Reply:
It can be more than Lsp or User, for example, search. We would then need to add to the `open_` functions an additional parameter to pass in this information, I think. What do you think?